src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1307,21 +1335,33 @@ sol_http_progressive_response_del(struct sol_http_progressive_response *progress
 
 SOL_API int
 sol_http_progressive_response_feed(struct sol_http_progressive_response *progressive,
-    const struct sol_str_slice data)
+    struct sol_blob *blob)
 {
     int ret;
+    size_t total;
 
     SOL_NULL_CHECK(progressive, -EINVAL);
     SOL_EXP_CHECK(progressive->delete_me == true, -EINVAL);
+    SOL_NULL_CHECK(blob, -EINVAL);
 
-    ret = sol_buffer_append_slice(&progressive->buffer, data);
+    ret = sol_util_size_add(progressive->accumulated_bytes, blob->size, &total);
     SOL_INT_CHECK(ret, < 0, ret);
 
+    if (progressive->tx_size && total >= progressive->tx_size)
+        return -ENOSPC;
+
+    ret = sol_ptr_vector_append(&progressive->pending_blobs, blob);
+    SOL_INT_CHECK(ret, < 0, ret);
+
+    sol_blob_ref(blob);
+
     if (progressive->request->suspended) {
         progressive->request->suspended = false;
         MHD_resume_connection(progressive->request->connection);
     }
 
+    progressive->accumulated_bytes += total;
```

Comment:
You add `progressive->accumulated_bytes` and `blob->size` to get `total`. Then you add `total` to `progressive->accumulated_bytes`... is it right? Shouldn't `progressive->accumulated_bytes` just receive `total`? The way you do, if `progressive->accumulated_bytes`is `10` and `blob->size` is `5`, `total` will be `15`. And after this line, `progressive->accumulated_bytes` will be `25`...